src/object/srv_obj_migrate.c

```diff
@@ -3271,8 +3272,70 @@ migrate_obj_ult(void *data)
 		}
 	}
 
+	if (!tls->mpt_reintegrating_check) {
+		struct pool_target *target;
+		struct ds_pool *pool = tls->mpt_pool->spc_pool;
+
+		ABT_rwlock_rdlock(pool->sp_lock);
+		rc = pool_map_find_target_by_rank_idx(pool->sp_map, dss_self_rank(),
+						      dss_get_module_info()->dmi_tgt_id,
+						      &target);
+		ABT_rwlock_unlock(pool->sp_lock);
+		D_ASSERT(rc == 1);
+
+		if (target->ta_comp.co_status == PO_COMP_ST_UP &&
+		    target->ta_comp.co_in_ver <= tls->mpt_version)
+			tls->mpt_reintegrating = 1;
+		D_DEBUG(DB_REBUILD, "status %u in version %u, mpt_reintegrating %d\n",
+			target->ta_comp.co_status, target->ta_comp.co_in_ver,
+			tls->mpt_reintegrating);
+
+		tls->mpt_reintegrating_check = 1;
+	}
+
+	if (tls->mpt_reintegrating) {
+		struct ds_cont_child *cont_child = NULL;
+
+		/* check again to see if the container is being destroyed. */
+		migrate_get_cont_child(tls, arg->cont_uuid, &cont_child, false);
+		if (cont_child != NULL && !cont_child->sc_stopping) {
+			if (vos_oi_exist(cont_child->sc_hdl, arg->oid)) {
+				rc = vos_dtx_cmt_reindex(cont_child->sc_hdl);
```

Comment:
It is unnecessary to reindex DTX if the opening container only for get the global stable epoch. On the other hand, since the global stable epoch is (at least) per-container, then we can pass it as one of the parameters for `migrate_obj_ult()` instead of loading it from VOS for every object again and again.

Reply:
seems you are checking an older version, the call to vos_dtx_cmt_reindex removed already.
migrate_get_cont_child() is the original code maybe useful to detect if the cont destroy during rebuild WIP.